### PR TITLE
FIX-#4584: Enable pdb debug when running cloud tests

### LINKED ISSUE #4584: Facilitate pdb usage in Modin-in-the-Cloud
**Is your feature request related to a problem? Please describe.**
Currently it's impossible to use `pdb` when debugging some failure on remote side of the RPyC tunnel, even if tests are running locally, because `pdb` requires stdin/stdout/stderr to be user-controllable, and current implementation of `LocalConnection` hides those either in `subprocess.PIPE` or in `subprocess.DEVNULL`.

### PR DESCRIPTION
**Pull request by vnlitvinov**
Signed-off-by: Vasily Litvinov <fam1ly.n4me@yandex.ru>

## What do these changes do?
Enable using `pdb.set_trace()` and stuff like that on remote side of cloud connection, to facilitate debugging cloud-related issues.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4584 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/experimental/cloud/local_cluster.py [hunk 1]
```diff
@@ -57,9 +61,9 @@ def _run(self, sshcmd: list, cmd: list, capture_out: bool = True):
             redirect.write(f"Running: {cmd}\n")
         return subprocess.Popen(
             cmd,
-            stdin=subprocess.DEVNULL,
-            stdout=redirect,
-            stderr=redirect,
+            stdin=None if _IS_PYTEST_DEBUG and not capture_out else subprocess.DEVNULL,
```
Comment (jeffreykennethli) [inline modin/experimental/cloud/local_cluster.py:64]: I'm not familiar with these test fixtures, so could you explain these lines? If you're running `pdb` (and `not capture_out`) why should `stdin`, `stdout`, and `stderr` be None?

**Review by devin-petersohn — DISMISSED** [review]
LGTM, will wait to merge until @jeffreykennethli questions are answered

File: modin/experimental/cloud/local_cluster.py (hunk 1 above)
Comment (vnlitvinov) [reply modin/experimental/cloud/local_cluster.py:64]: These are not fixtures, it's a normal `subprocess.Popen()` being called here.
And the reason for those to be `None` is so the "remote" end actually can be interacted with using std streams.

**Comment by vnlitvinov** [comment]
@jeffreykennethli @devin-petersohn can we approve and merge this?

**Comment by jeffreykennethli** [comment]
> @jeffreykennethli @devin-petersohn can we approve and merge this?

Makes sense! Ty for the explanations @vnlitvinov

File: modin/experimental/cloud/local_cluster.py [hunk 2]
```diff
@@ -22,6 +22,10 @@
 from .tracing.tracing_connection import TracingWrappingConnection
 from modin.config import DoTraceRpyc
 
+_IS_PYTEST_DEBUG = (
+    any(arg.endswith("pytest") for arg in sys.argv) and "--pdb" in sys.argv
```
Comment (YarShev) [inline modin/experimental/cloud/local_cluster.py:26]: Will `_IS_PYTEST_DEBUG` be True for these both cases?

```bash
$ python -m pytest --pdb ... # 1
$ pytest --pdb ... # 2
```

File: modin/conftest.py [hunk 3]
```diff
@@ -129,6 +129,18 @@ def simulate_cloud(request):
     if mode == "off":
         yield
         return
+    if (
+        request.config.getoption("usepdb")
+        and request.config.getoption("capture") != "no"
```
Comment (YarShev) [inline modin/conftest.py:134]: Should we check the presence of '-s' as well?

**Review by YarShev — COMMENTED** [review]
@vnlitvinov, left a couple of comments. Also, it is probably worth adding some documentation on this. What do you think?

**Review by devin-petersohn — COMMENTED** [review]
@vnlitvinov This looks pretty good to me, but there are a couple of questions from @YarShev. Will let @YarShev make the final approval.

File: modin/conftest.py (hunk 3 above)
Comment (vnlitvinov) [reply modin/conftest.py:134]: it's `capture` option 😉 checked at exactly the line you left the comment

**Comment by vnlitvinov** [comment]
> Also, it is probably worth adding some documentation on this. What do you think?

I think all this feature is rather niche (and I don't expect a lot of contributions here), and I think anyone who's going to dive into developing it would find this place easy enough.

That said, I'm too lazy to write a proper "cloud debugging guide" now, and just stating "run `pytest --pdb -s` if your simulated cloud tests are failing" isn't that useful IMHO (especially given `pytest --pdb` _without_ `-s` would tell about that option anyway).

**Review by YarShev — APPROVED** [review]
@vnlitvinov, LGTM, thanks!

**Pull request merged**